opus/application/static_media/js/browse.js

```diff
@@ -1196,6 +1200,53 @@ var o_browse = {
         return url;
     },
 
+    // check the cache of both rendered elements and variable o_browse.galleryData; remove 'far away' observations
+    // from cache to avoid buildup of too much data in the browser which slows things down
+    // Two functions; one to delete single elements, just a tool for the main one, manageObservationCache, to loop.
+    manageObservationCache: function(count, append) {
+        let tab = `#${opus.prefs.view}`;
+        var galleryObsElem =  $(`${tab} .gallery [data-obs]`);
+        var tableObsElem = $(`${tab} .op-dataTable-view [data-obs]`);
+
+        function deleteCachedObservation(index) {
+            // don't delete the metadata if the observation is in the cart
+            if (!galleryObsElem.eq(index).hasClass("in")) {
+                let delOpusId = galleryObsElem.eq(index).data("id");
+                delete o_browse.galleryData[delOpusId];
+            }
+            galleryObsElem.eq(index).remove();
+            tableObsElem.eq(index).remove();
+        };
+
+        let beginningIndex = galleryObsElem.first().index();
+        let endingIndex = galleryObsElem.last().index();
+        if (beginningIndex < 0 || endingIndex < 0) {
+            // this only happens when there are no elements rendered, so why bother...
+            // probably don't need to check both, but why not...
+            return;
+        }
+
+        let totalObs = endingIndex - beginningIndex;
+        if (totalObs + count > opus.maxBufferSize) {
+            // if we are appending, remove from the top
+            console.log(`before: ${Object.keys(o_browse.galleryData).length}`);
+            count = (count > endingIndex ? endingIndex : count);
+            if (append) {
+                // this is theoretically the faster way to delete lots of data, as jquery selector eval is slow
+                for (let index = 0; index < count ; index++) {
+                    deleteCachedObservation(index);
+                }
+            } else {
+                let deleteTo = endingIndex - count;
+                deleteTo = (deleteTo >= 0 ? deleteTo : 0);
```

Comment:
I think this has an off-by-one error. It should be endingIndex-count+1. That way if count=1, then we go from endingIndex to endingIndex and remove one element.

Math.max(endingIndex-count+1, 0) might be easier to read than using ?: